lang/src/ast.rs

```diff
@@ -352,36 +358,39 @@ impl FnArg {
 impl quote::ToTokens for FnArg {
     fn to_tokens(&self, tokens: &mut TokenStream2) {
         match self {
-            FnArg::SelfRef(arg_self_ref) => arg_self_ref.to_tokens(tokens),
-            FnArg::SelfValue(arg_self_value) => arg_self_value.to_tokens(tokens),
-            FnArg::Captured(arg_captured) => arg_captured.to_tokens(tokens),
+            FnArg::Receiver(receiver) => receiver.to_tokens(tokens),
+            FnArg::Typed(pat_ty) => pat_ty.to_tokens(tokens),
         }
     }
 }
 
 #[derive(Debug)]
-pub struct ArgCaptured {
-    pub pat: syn::Pat,
+pub struct PatType {
+    pub attrs: Vec<syn::Attribute>,
+    pub pat: Box<syn::Pat>,
     pub colon_token: Token![:],
-    pub ty: syn::Type,
+    pub ty: Box<syn::Type>,
```

Comment:
Why is `Box` required here?

Reply:
`PatType` and `Pat` are large enum variants. 
Just like [PatType](https://docs.rs/syn/1.0.8/syn/struct.PatType.html) and [Pat](https://docs.rs/syn/1.0.8/syn/enum.Pat.html) in syn.

Reply:
okay sounds fair